bodhi/server/consumers/masher.py

```diff
@@ -1005,24 +1005,30 @@ def _get_master_repomd_url(self, arch):
             basestring: A URL on the master mirror where the repomd.xml file should be synchronized.
         """
         release = self.compose.release.id_prefix.lower().replace('-', '_')
+        version = self.compose.release.version
         request = self.compose.request.value
 
+        # First check to see if there's an override for the current version, if not, fall back.
+        # This will first try fedora_28_stable_(suffix), and then fedora_stable_(suffix).
+        key_prefixes = ['%s_%s_%s' % (release, version, request),
+                        '%s_%s' % (release, request)]
         # If the release has primary_arches defined in the config, we need to consider whether to
         # use the release's *alt_master_repomd setting.
         primary_arches = config.get(
             '{release}_{version}_primary_arches'.format(
                 release=release, version=self.compose.release.version))
         if primary_arches and arch not in primary_arches.split():
-            key = '%s_%s_alt_master_repomd'
+            suffix = '_alt_master_repomd'
         else:
-            key = '%s_%s_master_repomd'
-        key = key % (release, request)
+            suffix = '_master_repomd'
 
-        master_repomd = config.get(key)
-        if not master_repomd:
-            raise ValueError("Could not find %s in the config file" % key)
+        keys = [key_prefix + suffix for key_prefix in key_prefixes]
 
-        return master_repomd % (self.compose.release.version, arch)
+        for key in keys:
+            val = config.get(key)
+            if val:
+                return val % (version, arch)
+        raise ValueError("Could not find any of %s in the config file" % ','.join(keys))
```

Comment:
[```ValueError```](https://docs.python.org/3/library/exceptions.html#ValueError) is used to indicate that an argument to a function or method is the right type but an unacceptable value. In this case, it's really that the settings file was missing a particular setting, not that the arch passed to this function was invalid.

It might make sense to raise an ```IndexError``` since the config didn't have any of the requested keys, or you could just do a more general ```RuntimeError```.

Reply:
We should add a test to assert that this ```ValueError``` is raised.

Reply:
https://github.com/puiterwijk/bodhi/blob/2efb5639c6d4d282cfe2ef90b697c9e223584ad5/bodhi/tests/server/consumers/test_masher.py#L1797

Reply:
Nevermind, I didn't notice that there already is a test for the ValueError, or even that it was already raised, so don't worry about the exception type since it was already like this.